### ISSUE TITLE
Do I need to generate the full data to reproduce your result in the README

### ISSUE THREAD

**Issue by MingSun-Tse** [twtygqyy/pytorch-vdsr#27]
Hi, thank you very much for sharing this code.

I git clone your code and run the `main_vdsr.py` to train. Everything is fine, except that, when I use the output model `checkpoint/model_epoch_50.pth`  to eval, the reported  `PSNR_predicted` is very close to `PSNR_bicubic`. 

Eval log is here:

```
In [15]: py eval.py --model checkpoint/model_epoch_50.pth --dataset Set5 --cuda
=> use gpu id: '0'
('Processing ', 'Set5_mat/woman_GT_x2.mat')
('Processing ', 'Set5_mat/butterfly_GT_x2.mat')
('Processing ', 'Set5_mat/bird_GT_x2.mat')
('Processing ', 'Set5_mat/baby_GT_x2.mat')
('Processing ', 'Set5_mat/head_GT_x2.mat')
('Scale=', 2)
('Dataset=', 'Set5')
('PSNR_predicted=', 33.69041184218558)
('PSNR_bicubic=', 33.69039381292539)
It takes average 0.0161561489105s for processing
('Processing ', 'Set5_mat/woman_GT_x3.mat')
('Processing ', 'Set5_mat/butterfly_GT_x3.mat')
('Processing ', 'Set5_mat/head_GT_x3.mat')
('Processing ', 'Set5_mat/baby_GT_x3.mat')
('Processing ', 'Set5_mat/bird_GT_x3.mat')
('Scale=', 3)
('Dataset=', 'Set5')
('PSNR_predicted=', 30.407706294827427)
('PSNR_bicubic=', 30.407692343235453)
It takes average 0.00243172645569s for processing
('Processing ', 'Set5_mat/bird_GT_x4.mat')
('Processing ', 'Set5_mat/woman_GT_x4.mat')
('Processing ', 'Set5_mat/baby_GT_x4.mat')
('Processing ', 'Set5_mat/butterfly_GT_x4.mat')
('Processing ', 'Set5_mat/head_GT_x4.mat')
('Scale=', 4)
('Dataset=', 'Set5')
('PSNR_predicted=', 28.41455707403013)
('PSNR_bicubic=', 28.41454827257395)
It takes average 0.00247440338135s for processing
```
My Environment: Ubuntu 1404, pytorch 0.4.1, python 2.7.

I noticed the training data is the `train.h5`, which is only 13MB. But based on the other issues, I guess the full data is far more than this. So is my problem due to the lack of training examples? (Sorry that I'm new to SR, knowing little about the commonly-used datasets etc.)

Thank you very much.

**Comment by MingSun-Tse** [comment]
After confirmation, that's exactly my problem...

After 50 epochs' training, here is the new result, for someone's reference if possible. Not as good as the result in the README, but close.
```
In [14]: py eval.py --model checkpoint/%s_model_epoch_192-20181019-0832.pth --dataset Set5 --cuda --gpu 5
=> use gpu id: '5'
Processing  Set5_mat/woman_GT_x2.mat
Processing  Set5_mat/butterfly_GT_x2.mat
Processing  Set5_mat/bird_GT_x2.mat
Processing  Set5_mat/baby_GT_x2.mat
Processing  Set5_mat/head_GT_x2.mat
Scale= 2
Dataset= Set5
PSNR_predicted= 37.556603288207
PSNR_bicubic= 33.69039381292539
It takes average 0.022900485992431642s for processing
Processing  Set5_mat/woman_GT_x3.mat
Processing  Set5_mat/butterfly_GT_x3.mat
Processing  Set5_mat/head_GT_x3.mat
Processing  Set5_mat/baby_GT_x3.mat
Processing  Set5_mat/bird_GT_x3.mat
Scale= 3
Dataset= Set5
PSNR_predicted= 33.76509299307266
PSNR_bicubic= 30.407692343235453
It takes average 0.0026937007904052736s for processing
Processing  Set5_mat/bird_GT_x4.mat
Processing  Set5_mat/woman_GT_x4.mat
Processing  Set5_mat/baby_GT_x4.mat
Processing  Set5_mat/butterfly_GT_x4.mat
Processing  Set5_mat/head_GT_x4.mat
Scale= 4
Dataset= Set5
PSNR_predicted= 31.414080321483095
PSNR_bicubic= 28.41454827257395
It takes average 0.002594184875488281s for processing
```

**Comment by Jee-King** [comment]
Hi.
Can you remember how much the loss can be reduced?

**Comment by MingSun-Tse** [comment]
I checked my training log. The training loss oscillates much. And during oscillation, the training loss decreases a little (from ~395 to ~365).

**Comment by MingSun-Tse** [comment]
> Hi.
> Can you remember how much the loss can be reduced?

I checked my training log. The training loss oscillates much. And during oscillation, the training loss decreases a little (from ~395 to ~365).

**Comment by Jee-King** [comment]
Thank you  for your reply.
Because I used python3 pytorch4.1, so I just use the vdsr.py (make network file).
I create 19808 images for training, and use same config.
Here is my training process:
![12](https://user-images.githubusercontent.com/24266511/53572226-c27b1680-3ba5-11e9-977a-ebe6c9d96f74.png)
after 150 epoch, the result is not good

**Comment by arvindsaripalli** [comment]
Hi @MingSun-Tse I'm trying to get the result that you did on the train.h5 dataset. There are 1000 downsampled 41x41 images in the dataloader so that seems correct. I ran training for 50 epochs for various batch sizes and had loss decrease to around 0.0000000214. My eval looks similar to your first one with not much improvement. Any suggestions?

**Comment by MingSun-Tse** [comment]
> Hi @MingSun-Tse I'm trying to get the result that you did on the train.h5 dataset. There are 1000 downsampled 41x41 images in the dataloader so that seems correct. I ran training for 50 epochs for various batch sizes and had loss decrease to around 0.0000000214. My eval looks similar to your first one with not much improvement. Any suggestions?

If your loss is so low, I guess there is something wrong with your data. I did nothing special but following the guidance of that README. You can check every step in it, you will be fine. Good luck!

**Comment by Jee-King** [comment]
Hi, sorry bother you again. Could you send me full log-file of your training ? This is my email, 1406011307@qq.com. Thank you very much!

**Comment by paulo-eusebio** [comment]
> 
> 
> Hi @MingSun-Tse I'm trying to get the result that you did on the train.h5 dataset. There are 1000 downsampled 41x41 images in the dataloader so that seems correct. I ran training for 50 epochs for various batch sizes and had loss decrease to around 0.0000000214. My eval looks similar to your first one with not much improvement. Any suggestions?

You probably have the MSE loss averaged over the batch, the code provided does not average the loss. To obtain similar loss values during training, use nn.mseloss(reduce='sum') which sums the loss for all batch sample. The default averages the loss, which is why you get low values.
